c7n/resources/route53.py

```diff
@@ -212,3 +214,107 @@ def process_resource_set(self, domains, keys):
             client.delete_tags_for_domain(
                 DomainName=d[self.id_key],
                 TagsToDelete=keys)
+
+
+@HostedZone.action_registry.register('enable-query-logging')
+class EnableQueryLogging(BaseAction):
+    """Enables query logging on a public hosted zone
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: enablednsquerylogging
+            resource: hostedzone
+            filters:
+                - type: query-logging-enabled
+                state: false
+            actions:
+                - type: enable-query-logging
+                state: true
+                logretentiondays: 30
+    """
+    permissions = ('route53:GetQueryLoggingConfig','route53:GetHostedZone',
+        'route53:CreateQueryLoggingConfig','route53:DeleteQueryLoggingConfig',
+        'logs:CreateLogGroup','logs:DescribeLogGroups','logs:PutRetentionPolicy')
+
+    schema = type_schema('enable-query-logging', state={'type': 'boolean'},
+        logretentiondays={'type': 'number'})
+
+    def process(self, resources):
+
+        client = local_session(self.manager.session_factory).client('route53')
+        state = self.data.get('state', False)
+        logretentiondays = self.data.get('logretentiondays', 30)
+        valid_days = [1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653]
```

Comment:
these values seem arbitrary, and aren't actually per any api requirements. ie if an org wants to permanently retain or do 32 days, its unclear why this validation is required.